src/EnergyPlus/ElectricBaseboardRadiator.cc

```diff
@@ -558,80 +523,62 @@ namespace ElectricBaseboardRadiator {
-        int NumElecBaseboards = state.dataElectBaseboardRad->NumElecBaseboards;
-        auto &ZeroSourceSumHATsurf = state.dataElectBaseboardRad->ZeroSourceSumHATsurf;
-        auto &QBBElecRadSource = state.dataElectBaseboardRad->QBBElecRadSource;
-        auto &QBBElecRadSrcAvg = state.dataElectBaseboardRad->QBBElecRadSrcAvg;
-        auto &LastQBBElecRadSrc = state.dataElectBaseboardRad->LastQBBElecRadSrc;
-        auto &LastSysTimeElapsed = state.dataElectBaseboardRad->LastSysTimeElapsed;
-        auto &LastTimeStepSys = state.dataElectBaseboardRad->LastTimeStepSys;
+        auto &elecBaseboard = state.dataElectBaseboardRad->ElecBaseboard(BaseboardNum);
 
         // Do the one time initializations
         if (state.dataElectBaseboardRad->MyOneTimeFlag) {
             // initialize the environment and sizing flags
-            state.dataElectBaseboardRad->MyEnvrnFlag.dimension(NumElecBaseboards, true);
-            state.dataElectBaseboardRad->MySizeFlag.dimension(NumElecBaseboards, true);
-            ZeroSourceSumHATsurf.dimension(state.dataGlobal->NumOfZones, 0.0);
-            QBBElecRadSource.dimension(NumElecBaseboards, 0.0);
-            QBBElecRadSrcAvg.dimension(NumElecBaseboards, 0.0);
-            LastQBBElecRadSrc.dimension(NumElecBaseboards, 0.0);
-            LastSysTimeElapsed.dimension(NumElecBaseboards, 0.0);
-            LastTimeStepSys.dimension(NumElecBaseboards, 0.0);
-
+            state.dataElectBaseboardRad->ZeroSourceSumHATsurf.dimension(state.dataGlobal->NumOfZones, 0.0);
             state.dataElectBaseboardRad->MyOneTimeFlag = false;
         }
 
-        if (!state.dataGlobal->SysSizingCalc && state.dataElectBaseboardRad->MySizeFlag(BaseboardNum)) {
+        if (!state.dataGlobal->SysSizingCalc && elecBaseboard.MySizeFlag) {
             // for each coil, do the sizing once.
             SizeElectricBaseboard(state, BaseboardNum);
-            state.dataElectBaseboardRad->MySizeFlag(BaseboardNum) = false;
+            elecBaseboard.MySizeFlag = false;
         }
 
         // Do the Begin Environment initializations
-        if (state.dataGlobal->BeginEnvrnFlag && state.dataElectBaseboardRad->MyEnvrnFlag(BaseboardNum)) {
+        if (state.dataGlobal->BeginEnvrnFlag && elecBaseboard.MyEnvrnFlag) {
             // Initialize
-            ZeroSourceSumHATsurf = 0.0;
-            QBBElecRadSource = 0.0;
-            QBBElecRadSrcAvg = 0.0;
-            LastQBBElecRadSrc = 0.0;
-            LastSysTimeElapsed = 0.0;
-            LastTimeStepSys = 0.0;
-
-            state.dataElectBaseboardRad->MyEnvrnFlag(BaseboardNum) = false;
+            state.dataElectBaseboardRad->ZeroSourceSumHATsurf = 0.0;
+            elecBaseboard.QBBElecRadSource = 0.0;
+            elecBaseboard.QBBElecRadSrcAvg = 0.0;
+            elecBaseboard.LastQBBElecRadSrc = 0.0;
+            elecBaseboard.LastSysTimeElapsed = 0.0;
+            elecBaseboard.LastTimeStepSys = 0.0;
+
+            elecBaseboard.MyEnvrnFlag = false;
```

Comment:
I agree.  The whole calling one component and it messes with all the components of that type was not ideal.  Through the plant refactor we cleaned up a bunch of these, sometimes with diffs sometimes not.  If they pop up, I am happy to run before/after of this one commit to confirm this is the cause.

Reply:
I think it's worth cleaning up these last few items. I'll put ZeroSourceSumHATsurf right next to sumHATsurf.

            state.dataElectBaseboardRad->ZeroSourceSumHATsurf(ControlledZoneNum) = state.dataHeatBal->Zone(ControlledZoneNum).sumHATsurf(state);

Reply:
The real question here is whether ZeroSourceSumHATsurf could simply be local - does it need to be stored in an array? This method is used by every piece of radiant equipment, so making is a zone member variable might introduce crosstalk if the same zone has more than one type of radiant equipment.

Reply:
Good point. The index to a zone was confusing on what this variable was used for. Maybe a zone with 2 baseboards would uncover something. Regardless of where this lives, this variable is the sumHAT of the zone (or space) surfaces. How would 2 radiant systems in that zone split/share this value? Moving this variable to the zone stuct does seem to be problematic since this variable was a member of each radiant system. I think moving this back is probably the right thing to do for now. A radiant system cannot serve more than 1 zone so making it local to each radiant system seems logical.

    struct ElecBaseboardParams
    {
        Array1D_string SurfaceName;
        Array1D_int SurfacePtr;
        int ZonePtr;